http-svr/http_test.ml

```diff
@@ -151,8 +152,49 @@ let _ =
         [
             "accept_simple" >:: test_accept_simple;
             "accept_complex" >:: test_accept_complex;
-			"radix1" >:: test_radix_tree1;
-			"radix2" >:: test_radix_tree2;
-			"test_url" >:: test_url
+            "radix1" >:: test_radix_tree1;
+            "radix2" >:: test_radix_tree2;
+            "test_url" >:: test_url;
+            "test_read_http_request_header" >:: test_read_http_request_header;
 		] in
     run_test_tt ~verbose:!verbose suite
+
+let with_fd input f =
+  let read_fd, write_fd = Unix.pipe () in
+  let _:int = Unix.write write_fd (Bytes.of_string input) 0 (String.length input) in
+  Fun.protect ~finally:(fun () -> Unix.close read_fd ; Unix.close write_fd) (fun () -> f read_fd)
+
+let cross xs ys zs =
+  xs |> List.fold_left (fun acc x ->
+  ys |> List.fold_left (fun acc y ->
+  zs |> List.fold_left (fun acc z -> (x,y,z)::acc) acc) acc) []
+
+let test_read_http_request_header _ =
```

Comment:
This function is tried to be used on line 158, which should fail because it means it hasn't been bound. On the original commit this block happens above the definition of the tests.